lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java

```diff
@@ -506,14 +493,55 @@ void should_DeleteOldHistoryCleanupJobs_When_InitializingSchedule() throws Excep
     return DynamicTest.stream(iterator, c -> "for parentBusinessProcessId = '" + c + "'", test);
   }
 
+  private TaskHistoryEvent createTaskHistoryEvent(
+      String workbasketKey,
+      String taskId,
+      String type,
+      String previousWorkbasketId,
+      String userid,
+      String details) {
+    TaskHistoryEvent historyEvent = new TaskHistoryEvent();
+    historyEvent.setId(IdGenerator.generateWithPrefix(IdGenerator.ID_PREFIX_TASK_HISTORY_EVENT));
+    historyEvent.setUserId(userid);
+    historyEvent.setDetails(details);
+    historyEvent.setWorkbasketKey(workbasketKey);
+    historyEvent.setTaskId(taskId);
+    historyEvent.setEventType(type);
+    historyEvent.setOldValue(previousWorkbasketId);
+    return historyEvent;
+  }
+
   private void createTaskanaEngineWithNewConfig(
```

Comment:
This might not work combined with other @TaskanaInject Objects because of the test api logic. We should find another solution

Reply:
I do not understand, what does not work here?

Reply:
I have changed it to use nested classes and use modify function to change the property